lib/promscrape/discovery/kubernetes/pod.go

```diff
@@ -251,6 +251,13 @@ func (p *Pod) appendCommonLabels(m *promutil.Labels, gw *groupWatcher) {
 			n.Metadata.registerLabelsAndAnnotations("__meta_kubernetes_node", m)
 		}
 	}
+	if gw.attachNamespaceMetadata {
+		o := gw.getObjectByRoleLocked("namespace", "", p.Metadata.Namespace)
```

Comment:
here and in other instances, I chose to call `gw.getObjectByRole` with an explicit empty string "" for the namespace, as the namespace is a cluster-scoped resource.
(nodes also are cluster-scoped, but for nodes the namespace was specified, which is not useful as it's set to the empty string in `gw.getObjectByRole` function)
but I can use the same style if you prefer me to